### PR TITLE
Feature/data discovery template

### PR DESCRIPTION
**Pull request by KoflerStefan**
template for deploying cube-api, db and data-to-cube funtionapp

### CODE REVIEW

**Review by florian-fischer-swarm — COMMENTED** [review]
In general looks really nice.
Once question: Which sku is chosen as DB server?

**Comment by KoflerStefan** [comment]
@florian-fischer-swarm: the db-server does not even have the option to choose a sku via the template. The elastic pool has an sku to choose from, but i actually do not know which is default here (seemingly depends on the region)

**Comment by KoflerStefan** [comment]
i think in our test it was 2 vCores but could be changed easily

**Comment by michael-rogger-swarm** [comment]
would be great to change to the smallest available option for the elastic pool.
Screenshot 2021-03-04 at 11 22 25
I guess it is this one?

**Comment by KoflerStefan** [comment]
okay updated it to deploy the cheapest model for elastic pool

File: data-discovery-deployment-template.JSON [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+      }
+    },
+    "applicationResourceGroup": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().name]",
+      "metadata": {
+        "description": "Resource group for the created resources."
+      }
+    },
+    "databaseServer": {
```
Comment (michael-rogger-swarm) [inline data-discovery-deployment-template.JSON:26]: Screenshot 2021-03-04 at 12 41 46
Screenshot 2021-03-04 at 12 42 18
Hmm did I specify the pool wrong? 🤔

**Review by michael-rogger-swarm — CHANGES_REQUESTED** [review]

File: data-discovery-deployment-template.JSON (hunk 1 above)
Comment (KoflerStefan) [reply data-discovery-deployment-template.JSON:26]: hard to say like that, do you have a screenshot of the resources of the resource-group were the database and the elastic pool are located in?

File: data-discovery-deployment-template.JSON [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+      }
+    },
+    "sccAppTenant": {
+      "type": "string",
+      "metadata": {
+        "description": "The App Tenant of the Swarm Control Center"
+      }
+    }
+  },
+  "variables": {
```
Comment (florian-fischer-swarm) [inline data-discovery-deployment-template.JSON:139]: Pretty nice :-)

File: data-discovery-deployment-template.JSON [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+            {
+              "name": "CUBEJS_DB_USER",
+              "value": "[parameters('databaseServerAdminName')]"
+            },
+            {
+              "name": "CUBEJS_DB_PASS",
+              "value": "[parameters('databaseServerAdminPassword')]"
+            },
+            {
+              "name": "CUBEJS_DEV_MODE",
```
Comment (florian-fischer-swarm) [inline data-discovery-deployment-template.JSON:372]: Ooops, that is the example I put for running the dockerfile locally for development into the README, sorry. Needs to be "false" or just removed (has a default) for real deployments.

**Review by florian-fischer-swarm — CHANGES_REQUESTED** [review]
Just minor change wrt dev mode of cubejs.

File: data-discovery-deployment-template.JSON (hunk 3 above)
Comment (KoflerStefan) [reply data-discovery-deployment-template.JSON]: already removed 👍

**Review by florian-fischer-swarm — APPROVED** [review]

**Review by florian-fischer-swarm — APPROVED** [review]
image

**Review by michael-rogger-swarm — APPROVED** [review]
👍

**Pull request merged**